src/content/docs/browser/new-relic-browser/browser-apis/setuserid.mdx

```diff
@@ -0,0 +1,79 @@
+---
+title: setUserId
+type: apiDoc
+shortDescription: Adds a user-defined identifier string to subsequent events on the page.
+tags:
+  - Browser
+  - Browser monitoring
+  - Browser agent and SPA API
+metaDescription: Browser API call to add an identifier to subsequent events on the page.
+redirects:
+---
+
+<Callout variant="important">
+This API will work for any browser edition (Browser Lite, Pro, or Pro+SPA).
+</Callout>  
+
+## Syntax
+
+```js
+newrelic.setUserId(value: string|null)
+```
+
+Adds a user-defined identifier string to subsequent events on the page.
+
+## Requirements
+
+Agent version [1.230.0](/docs/release-notes/new-relic-browser-release-notes/browser-agent-release-notes) or higher.
+
+## Description
+
+Upon executing this function with a valid value, the browser agent records the value as the `enduser.id` attribute with all events until the attribute is manually unset. The identifier will be stored in the browser, so that subsequent page visits of the _same_ origin **within a session** span retain it on events. Do note that this functionlity may fluctuate depending on end-user browser privacy settings. If this function is called with a `value = null`, any existing user ID will be deleted from **both** the current page's events and the storage.
```

Comment:
@cwli24 I stumbled over this sentence: "The identifier will be stored in the browser, so that subsequent page visits of the _same_ origin **within a session** span retain it on events."

If this is accurate, could we use this revision: "The identifier will be stored in the browser, so that subsequent page visits retain it on events of the _same_ origin **within a session** span."

For the final sentence, I think you can drop "Do" like this: "Note that this functionality may fluctuate depending on end-user browser privacy settings. If this function is called with a `value = null`, any existing user ID will be deleted from **both** the current page's events and the storage"

Reply:
"Same origin" describes the page and cannot describe events.
"within a session" describes the page and can describe events though.